macOS/clang patches

kevinshome opened the pull request:
These are just a couple patches that needed to be implemented in order to successfully build (with `make bootstrap`) binfetch on macOS.

kevinshome committed:
basic macOS build patch

kevinshome committed:
patch ib bootstrap for macOS/clang

kevinshome committed:
create build directory on `make bootstrap`

kevinshome commented:
Also grouped in is a generic patch for `make bootstrap`, which adds the line `mkdir -p build` at the end of the `build_ib` make instruction.

kevinshome committed:
change prefix to `/usr/local` if macOS

Nik-Nothing reviewed (commented):
Hi!, Im very excited to see your pullrequest!, I do see a couple of things which are not correct though, here are some of my notes

-prefix should be /usr/local by default(that was my mistake) making the macos exception redundant
-comment on args.c.ib is not needed
-adding a seperate path for /usr/local/share is not the most flexible route, fine for now though, I just need to seperate destdir and prefix and then I can define the path to share as prefix with the preprocessor
-extra braces in ib.c should not be needed, and ib.c is not a part of this project

Commit:
adapted (see comment)

kevinshome commented:
Hey, happy to help how I can, and yeah, it was really just some messy patches to build lol, however, I'm still experiencing a couple issues with the build, which are still fixed by patches I added (idk if they could be implemented in a different way, just saying). They are:

> extra braces in ib.c should not be needed, and ib.c is not a part of this project

If you consider ib.c as part of the [ibranching](https://github.com/Nik-Nothing/ibranching) project, and it should be fixed as part of that, totally understand. However, the reason I added them was because of this issue during compilation (clang-1205.0.22.11):

```
./src/ib/ib.c:847:5: error: expected expression
                                char invalid[1];
                                ^
./src/ib/ib.c:848:5: error: use of undeclared identifier 'invalid'
                                invalid[0] = arg[i];
                                ^
./src/ib/ib.c:850:28: error: use of undeclared identifier 'invalid'
                                warn("invalid option", invalid, 0);
```

Like I said, if you'd rather this be a fix to [ibranching](https://github.com/Nik-Nothing/ibranching), totally understand. However, the only fix I've found to properly compile `ib.c`, even after your patch, is to add the braces.

> comment on args.c.ib is not needed

The reason I added the comment is because of this transpilation error:

```
 CC      build/args.c
build/args.c:67:12: error: label at end of compound statement: expected statement
                        default:
                                ^
                                 ;
```

which seems to originate from the [terminate() function](https://github.com/Nik-Nothing/ibranching/blob/main/ib.c.ib#L230) in [ibranching](https://github.com/Nik-Nothing/ibranching) not detecting it as a valid line that needs semi-colon termination. So the comment was simply a rough patch to get around that. Again, if you'd rather it be a patch to [ibranching](https://github.com/Nik-Nothing/ibranching), totally understand.

These are the only two build issues I seem to still be having currently. Just thought I would bring them to your attention and explain more in-depth why I made those changes, so hopefully we can figure out a solution together. Thank you again, and as a side note, I think this is a great project and hope it sees similar fame and usage to neofetch! :)

Nik-Nothing commented:
Hey @kevinshome !

I see now why you did that, I was able to reproduce with clang on linux, seams to be something with clangs handeling of case labels, I also realized though that I messed up in putting a very much outdated version of ib into bootstrap, the updated version doesn't have the same issue I also added a break to args.c.ib to fix the other label related issue

thank you very much for taking the time to elaborate :), and thank you for taking the time to report and your friendly wishes! <3

kevinshome commented:
@Nik-Nothing just tested, compiles perfectly, no issue! macOS compat seems solved :)
